backend/apps/volontulo/views/api.py

```diff
@@ -65,6 +69,62 @@ def login_view(request):
     )
 
 
+@api_view(['POST'])
+@authentication_classes((CsrfExemptSessionAuthentication,))
+@permission_classes((AllowAny,))
+def register_view(request):
+    """REST API register view."""
+    if not request.user.is_authenticated():
+        email = request.data.get('email')
+        password = request.data.get('password')
+
+        try:
+            user = User.objects.create_user(
+                username=email,
+                email=email,
+                password=password,
+                is_active=False
+            )
+
+            user.save()
+            profile = UserProfile(user=user)
+            ctx = {'token': profile.uuid, 'angular_root':
+                   settings.ANGULAR_ROOT}
+            profile.save()
+            send_mail(request, 'registration', [user.email], context=ctx)
+        except IntegrityError:
+            pass
+
+        return Response(
+            status=status.HTTP_201_CREATED,
+        )
+
+    return Response(
+        serializers.UserSerializer(request.user, context={
+            'request': request
+        }).data,
```

Comment:
Not sure what does this `data` actually produce, but aren't we more interested in telling the API consumer what's the problem with her request, instead of just returning the serialized user?

BTW, since it handles an attempt to create the user which already exists, you might consider 409 "Conflict" response.

BTW2, from the security point of view - is it intended to reveal that the account exists? This way the potential attacker might enumerate the existing accounts. I'm not sure if it's an issue in case of this very project, but I would be careful about that.

Reply:
As I understand these lines, You will get 201 when user alread exists, so it will not be reveal to potential attacker - 400 You will get only when You try to create user being already logged in.